src/components/AccordionMenu/index.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import makeClassName from 'classnames';
+
+import { ApplicationState } from '../../reducers';
+import { ConnectedReduxProps } from '../../configureStore';
+import { actions, isExpanded } from '../../reducers/accordionMenu';
+import { AnyReactNode } from '../../typeUtils';
+import styles from './styles.module.scss';
+
+export const makeItemContentId = (title: string) => {
+  const id = title.replace(/[^A-Za-z0-9]+/g, '');
+  if (!id) {
+    throw new Error(`An ID could not be generated from this title: "${title}"`);
+  }
+  return `accordionContent${id}`;
+};
+
+export type PublicItemProps = {
+  children: AnyReactNode;
+  title: string;
+};
+
+export type DefaultItemProps = {
+  expanded: boolean;
+};
+
+type ItemPropsFromState = {
+  expandedFromState: boolean;
+};
+
+type ItemProps = PublicItemProps &
+  DefaultItemProps &
+  ItemPropsFromState &
+  ConnectedReduxProps;
+
+export class AccordionItemBase extends React.Component<ItemProps> {
+  static defaultProps = {
+    expanded: false,
```

Comment:
I don't think there's a basic test for this. I.e., a test that renders an item without specifying a value for the `expanded` property, and asserts that `actions.toggleItem` was *not* dispatched.

Reply:
Good catch